fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/LakeTableTieringManager.java

```diff
@@ -0,0 +1,252 @@
+        // we put the table to a timout timer,
+        // if no liveness for the table in this period of time, we consider the tiering is
+        // invalid and put it back to wait to be scheduled by other tiering service
+        lakeTieringScheduleTimer.add(
+                new TimerTask(TIERING_SERVICE_TIMEOUT_MS) {
+                    @Override
+                    public void run() {
+                        inLock(
+                                lock,
+                                () -> {
+                                    // the table is not in tiering, we put it back to wait to be
+                                    // scheduled
+                                    if (!liveTieringTableIds.remove(tableId)) {
+                                        scheduleTableTiering(tableId);
+                                    } else {
+                                        // schedule liveness probe again
+                                        tieringLivenessProbe(tableId);
+                                    }
+                                });
+                    }
+                });
+    }
+
+    public void finishTableTiering(long tableId) {
+        inLock(
+                lock,
+                () -> {
+                    liveTieringTableIds.remove(tableId);
+                    scheduleTableTiering(tableId);
+                });
+    }
+
+    public void renewTieringLiveness(long tableId) {
+        inLock(
+                lock,
+                () -> {
+                    liveTieringTableIds.add(tableId);
+                });
+    }
+
+    @Override
+    public void close() throws Exception {
+        expirationReaper.initiateShutdown();
+        // improve shutdown time by waking up any ShutdownableThread(s) blocked on poll by
+        // sending a no-op.
+        lakeTieringScheduleTimer.add(
+                new TimerTask(0) {
+                    @Override
+                    public void run() {}
+                });
+        try {
+            expirationReaper.awaitShutdown();
+        } catch (InterruptedException e) {
+            throw new FlussRuntimeException("Error while shutdown delayed operation manager", e);
+        }
+
+        lakeTieringScheduleTimer.shutdown();
+    }
+
+    private class DelayedTiering extends TimerTask {
+
+        private final long tableId;
+
+        public DelayedTiering(long tableId, long delayMs) {
+            super(delayMs);
+            this.tableId = tableId;
+        }
+
+        @Override
+        public void run() {
+            inLock(
+                    lock,
+                    () -> {
+                        if (!liveTieringTableIds.contains(tableId)) {
+                            pendingTieringTables.add(tableId);
+                        }
+                    });
+        }
+    }
+
+    private class ExpiredOperationReaper extends ShutdownableThread {
+
+        public ExpiredOperationReaper() {
+            super("ExpiredOperationReaper", false);
```

Comment:
"LakeTieringExpiredOperationReaper" to distinguish with the name of `DelayedOperationManager.ExpiredOperationReaper`.